src/slick.grid.ts

```diff
@@ -3806,14 +3806,28 @@ export class SlickGrid<TData = any, C extends Column<TData> = Column<TData>, O e
 
     const frozenRowOffset = this.getFrozenRowOffset(row);
 
-    const rowHtml = `<div class="ui-widget-content ${rowCss}" data-top="${(this.getRowTop(row) - frozenRowOffset)}px">`; //style="top:${(this.getRowTop(row) - frozenRowOffset)}px"
+    const rowDiv = document.createElement('div');
+    let rowDivR: HTMLElement | undefined;
+    if (!this._options.nonce) {
+      const rowHtml = `<div class="ui-widget-content ${rowCss}" style="top: ${(this.getRowTop(row) - frozenRowOffset)}px;">`;
+      (stringArrayL as string[]).push(rowHtml);
 
-    stringArrayL.push(rowHtml);
-
-    if (this.hasFrozenColumns()) {
-      stringArrayR.push(rowHtml);
+      if (this.hasFrozenColumns()) {
+        (stringArrayR as string[]).push(rowHtml);
+      }
+    } else {
+      rowDiv.className = 'ui-widget-content ' + rowCss;
+      rowDiv.style.top = `${(this.getRowTop(row) - frozenRowOffset)}px`;
+      (stringArrayL as HTMLElement[]).push(rowDiv);
+      if (this.hasFrozenColumns()) {
+        //it has to be a deep copy otherwise we will have issues with pass by reference in js since
+        //attempting to add the same element to 2 different arrays will just move 1 item to the other array
+        rowDivR = rowDiv.cloneNode(true) as HTMLElement;
```

Comment:
cloneNode is needed to split unbind the elements from eachother, wasnt an issue before because it was just string which got transformed into elements.

example of the issue can be seen if you run this code in a console
```
var testDiv = document.createElement('div');
    var testDivArray = [testDiv];
    let x = document.createElement('div');
    let xRight = document.createElement('div');
    var elm;
    for (let i = 0; i < testDivArray.length; i++) {
      elm = testDivArray[i];
      x.appendChild(elm)
    }
    var elmR;
    for (let ii = 0; ii < testDivArray.length; ii++) {
      elmR = testDivArray[ii];
      xRight.appendChild(elmR)
    }
    console.log(x.children, xRight.children)
```

Reply:
oh so you managed to have it converted to pure HTML, great but when I tried it last week I had issues with frozen grids which I tried to do in PR #894 and then reverted my code when noticing too many problems, can you please try it with frozen grid. For this code, I don't think we need to keep the html string alternative, I would prefer to keep only pure HTML in this case even if there's a small perf hit. All grids should work with this code in place

Good point about the clone, I didn't think of that and that might have been my original problem with frozen grid

Reply:
frozen grid issues came because of the code example i gave, since instead of it copying the element to the frozen rows items it moved the item entirely, I've already tested it with frozen grid since that's where i also ran into issues, so before i commited any code i would at least have that working.

Frozen grids should work at least from what I've tested, no idea if they have a special case, but all the test case examples works